source/slang/slang-emit.cpp

```diff
@@ -1039,6 +1039,7 @@ Result linkAndOptimizeIR(
         case CodeGenTarget::MetalLib:
         case CodeGenTarget::MetalLibAssembly:
             byteAddressBufferOptions.scalarizeVectorLoadStore = true;
+            byteAddressBufferOptions.treatGetEquivalentStructuredBufferAsGetThis = true;
```

Comment:
What does treatGetEquivalentBufferAsGetThis do?

Reply:
`treatGetEquivalentStructuredBufferAsGetThis` causes all calls to `getEquivlentStructuredBuffer` to return a byteAddressBuffer (`this`) instead of a StructuredBuffer. 

This works since Metal does not care for the difference of these types (from what I understand).

The problem this solves is that the change allows legalizing and removing all calls to `getEquivlentStructuredBuffer` without introducing another duplicate buffer that DCE won't clean up.

Reply:
Can you add a comment to explain this?